internal/database/database.go

```diff
@@ -152,9 +152,23 @@ func (dbs *SQLdb) getFiles(datasetID string) ([]*FileInfo, error) {
 	files := []*FileInfo{}
 	db := dbs.DB
 
-	const query = "SELECT a.file_id, dataset_id, display_file_name, file_name, file_size, " +
-		"decrypted_file_size, decrypted_file_checksum, decrypted_file_checksum_type, file_status from " +
-		"local_ega_ebi.file a, local_ega_ebi.file_dataset b WHERE dataset_id = $1 AND a.file_id=b.file_id;"
+	const query = `
+		SELECT files.stable_id AS id,
+			datasets.stable_id AS dataset_id,
+			reverse(split_part(reverse(files.submission_file_path::text), '/'::text, 1)) AS display_file_name,
```

Comment:
It would solve https://github.com/neicnordic/sda-download/issues/229 if we would also get the full submission path here. We could put it into a new key in the JSON response to keep the API backwards compatible. Then the directory structure can be used optionally, if the client wants to.

Reply:
I have a second PR coming along that adds S3 endpoints - and another thing it does is adding a full-path field as well :) I tried splitting this part out since it was "just" updating the queries without changing the functionality.